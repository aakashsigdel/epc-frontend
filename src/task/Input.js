import React from 'react';
import PropTypes from 'prop-types';

const Input = props =>
  <input
    type="text"
    placeholder="Type on me"
    onChange={props.onChangeText}
  />;

Input.propTypes = {
  onChangeText: PropTypes.func.isRequired
};

export default Input;

