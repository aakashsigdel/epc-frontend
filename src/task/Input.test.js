import React from 'react';
import Input from './Input';

import renderer from 'react-test-renderer';

it('should render <Input /> correctly', () => {
  const component = renderer.create(<Input onChangeText={() => null}/>);
  let snapshotTree = component.toJSON();
  expect(snapshotTree).toMatchSnapshot();

  snapshotTree.props.onChange();
  snapshotTree = component.toJSON();
  expect(snapshotTree).toMatchSnapshot();
});


