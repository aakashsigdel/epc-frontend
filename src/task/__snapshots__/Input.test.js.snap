// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render <Input /> correctly 1`] = `
<input
  onChange={[Function]}
  type="text"
/>
`;

exports[`should render <Input /> correctly 2`] = `
<input
  onChange={[Function]}
  type="text"
/>
`;
